### PR TITLE
chore: update to electron 30

### LINKED ISSUE #214394: Electron 30 update
https://www.electronjs.org/blog/electron-30-0

### LINKED ISSUE #215924: Code 1.90.1 crashes on startup
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.90.1
- OS Version:  4.18.0-553.5.1.el8_10.x86_64, running in an virtual environment 

Steps to Reproduce:

1. Updated VSCode to the current version 1.90.1
2. Start with verbose `code --verbose`
3. VSCode crashes with the following message:

```shell
[0617/085433.786964:ERROR:elf_dynamic_array_reader.h(64)] tag not found
[0617/085433.789584:ERROR:file_io_posix.cc(145)] open /sys/devices/system/cpu/cpu0/cpufreq/scaling_cur_freq: No such file or directory (2)
[0617/085433.789605:ERROR:file_io_posix.cc(145)] open /sys/devices/system/cpu/cpu0/cpufreq/scaling_max_freq: No such file or directory (2)
[0617/085433.791755:ERROR:directory_reader_posix.cc(43)] opendir /net/machine/home/user/Development/vscode-1.90.1/userdata/Crashpad/attachments/bf6a73bb-efa8-4200-9654-29274ccf1a6c: No such file or directory (2)
```
Using the version 1.88.1, everything works.

Best regards,
Frank

### PR DESCRIPTION
**Pull request by deepak1556**
Fixes https://github.com/microsoft/vscode/issues/214394
Fixes https://github.com/microsoft/vscode/issues/215924

### CODE REVIEW

**Comment by deepak1556** [comment]
@chrmarti please review https://github.com/microsoft/vscode/pull/215406/commits/e4dd212002390c123d1f0ecff5e7516d2e715ae5

@connor4312 please review https://github.com/microsoft/vscode/pull/215406/commits/6c1649ebdbee93223d0171966bdc9f3a811706b7

@bpasero for rest of the files.

File: build/azure-pipelines/alpine/cli-build-alpine.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ steps:
     inputs:
       versionSource: fromFile
       versionFilePath: .nvmrc
-      nodejsMirror: https://github.com/joaomoreno/node-mirror/releases/download
```
Comment (deepak1556) [inline build/azure-pipelines/alpine/cli-build-alpine.yml:19]: Addressed in https://github.com/joaomoreno/node-mirror/pull/1

**Comment by connor4312** [comment]
lgtm

**Review by bpasero — DISMISSED** [review]

**Review by chrmarti — DISMISSED** [review]
LGTM!

**Review by bpasero — DISMISSED** [review]

**Comment by deepak1556** [comment]
PR is ready for merge, but I will merge it tomorrow since the new dependency has not been cached yet by the registry mirror.

**Review by bpasero — APPROVED** [review]

**Pull request merged**